### PR TITLE
ECOM-5009

### PR DESCRIPTION
**Pull request by awais786**
ECOM-5009
PR contains admin changes for program model.
Customize page for linking the program with excluded course runs.
Test cases.

### CODE REVIEW

File: course_discovery/templates/metadata/course_run.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                   <label for="{{ field.name }}">{{ field.label }}</label>
+                    <div class="form-row">
+                        {{ field }}
+                    </div>
+                </div>
+            {% endfor %}
+    </fieldset>
+    {% block submit_buttons_bottom %}
+        <div class="submit-row">
+                <a href="{% url 'admin:course_metadata_program_changelist' %}"><input type="button" name="Cancel" value="Cancel"></a>
```
Comment (waheedahmed) [inline course_discovery/templates/metadata/course_run.html]: Add `trans` on value.

**Comment by waheedahmed** [comment]
One comment otherwise LGTM 👍

**Comment by awais786** [comment]
@bderusha I have divided this ticket into 2 PRs. This is first PR which contains no autocomplete.  This contains custom page for course selections. Followup PR will bring the autocomplete with tests cases.

**Comment by awais786** [comment]
@bderusha please review this PR.

File: course_discovery/apps/course_metadata/tests/test_admin.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    def test_page_loads_only_course_related_runs(self):
+        """ Verify that course selection page loads only all course runs. Also marked checkboxes with
+        excluded courses runs only.
+        """
+        response = self.client.get(reverse('metadata:course_run_selection', args=(self.program.id,)))
+        html = '<input checked="checked" id="id_excluded_course_runs_0" '
+        html += 'name="excluded_course_runs" type="checkbox" value="{id}" />'.format(
+            id=self.excluded_course_run.id
+        )
+        self.assertEqual(response.status_code, 200)
```
Comment (bderusha) [inline course_discovery/apps/course_metadata/tests/test_admin.py]: assertContains already checks in the status code is 200 so you can remove this

File: course_discovery/apps/course_metadata/urls.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+"""
+URLs for the admin autocomplete lookups.
+"""
+from django.conf.urls import url
+
+from course_discovery.apps.course_metadata.views import CourseRunSelection
+
+
+urlpatterns = [
+    url(r'^course_runs/(?P<pk>\d+)/$', CourseRunSelection.as_view(), name='course_run_selection',),
```
Comment (bderusha) [inline course_discovery/apps/course_metadata/urls.py]: Is this an admin page?  If so can we have the URL structure match closer to that of the other admin URLs?

File: course_discovery/templates/metadata/course_run.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{% extends 'admin/base_site.html' %}
```
Comment (bderusha) [inline course_discovery/templates/metadata/course_run.html]: Same with this.  Let's make the file path to this clear that this is an admin template

File: course_discovery/templates/metadata/course_run.html (hunk 1 above)
Comment (bderusha) [inline course_discovery/templates/metadata/course_run.html]: 👍

File: course_discovery/urls.py [hunk 5]
```diff
@@ -42,6 +42,7 @@
         )
     ),
     url(r'^comments/', include('django_comments.urls')),
+    url(r'^metadata/', include('course_discovery.apps.course_metadata.urls', namespace='metadata')),
```
Comment (bderusha) [inline course_discovery/urls.py]: Add this to the `admin/` site urls above

**Comment by awais786** [comment]
@bderusha Please review this PR again.

**Comment by bderusha** [comment]
👍 squash 'n merge!

File: course_discovery/apps/course_metadata/forms.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class ProgramAdminForm(forms.ModelForm):
+
+    class Meta:
+        model = Program
+        exclude = (
+            'subtitle', 'category', 'type', 'marketing_slug', 'weeks_to_complete',
+            'min_hours_effort_per_week', 'max_hours_effort_per_week',
+            'authoring_organizations',
```
Comment (bderusha) [inline course_discovery/apps/course_metadata/forms.py:14]: @awais786 why are we excluding this?  Looks like it is included in the ACs https://openedx.atlassian.net/browse/ECOM-5009

**Comment by awais786** [comment]
@bderusha I am going to add in a new PR.

**Pull request merged**